File: admin/get cli/e_bandobas/lib/app/jsondata/PoliceStation.dart
class PoliceStation {
  PoliceStation({
    this.id,
    this.district,
    this.districtInGuj,
    this.taluko,
    this.talukoInGuj,
    this.contactNumber,
    this.address,
    this.policeStationName,
    this.policeStationNameInGujarati,
    this.headPolice,
  });

  PoliceStation.fromJson(dynamic json) {
    id = json['id'];
    district = json['district'];
    districtInGuj = json['districtInGuj'];
    taluko = json['taluko'];
    talukoInGuj = json['talukoInGuj'];
    contactNumber = json['contactNumber'];
    address = json['address'];
    policeStationName = json['policeStationName'];
    policeStationNameInGujarati = json['policeStationNameInGujarati'];
    headPolice = json['headPolice'];
  }
  num? id;
  String? district;
  String? districtInGuj;
  String? taluko;
  String? talukoInGuj;
  String? contactNumber;
  String? address;
  String? policeStationName;
  String? policeStationNameInGujarati;
  dynamic headPolice;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['id'] = id;
    map['district'] = district;
    map['districtInGuj'] = districtInGuj;
    map['taluko'] = taluko;
    map['talukoInGuj'] = talukoInGuj;
    map['contactNumber'] = contactNumber;
    map['address'] = address;
    map['policeStationName'] = policeStationName;
    map['policeStationNameInGujarati'] = policeStationNameInGujarati;
    map['headPolice'] = headPolice;
    return map;
  }
}

File: gujarat_police_backend/src/main/java/com/shyam/gujarat_police/entities/PoliceStation.java
package com.shyam.gujarat_police.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Pattern;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class PoliceStation extends BaseModel {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long id;

    @JsonIgnore
    @OneToMany(mappedBy="policeStation", cascade = CascadeType.PERSIST)
    private List<Police> police;

    @NotEmpty(message = "{validation.name.NotEmpty}")
    private String district;

    @NotEmpty(message = "{validation.name.NotEmpty}")
    private String districtInGuj;

    private String taluko;

    private String talukoInGuj;

    @Pattern(regexp="(^$|[0-9]{10})", message = "{validation.name.Size}")
    private String contactNumber;

//    @NotEmpty(message = "{validation.name.NotEmpty}")
    private String address;

    @NotEmpty(message = "{validation.name.NotEmpty}")
    private String policeStationName;

    @NotEmpty(message = "{validation.name.NotEmpty}")
    private String policeStationNameInGujarati;

    @Override
    public String toString() {
        return "PoliceStation{" +
                "id=" + id +
                ", district='" + district + '\'' +
                ", districtInGuj='" + districtInGuj + '\'' +
                ", taluko='" + taluko + '\'' +
                ", talukoInGuj='" + talukoInGuj + '\'' +
                ", contactNumber='" + contactNumber + '\'' +
                ", address='" + address + '\'' +
                ", policeStationName='" + policeStationName + '\'' +
                ", policeStationNameInGujarati='" + policeStationNameInGujarati + '\'' +
                '}';
    }

    @OneToOne
    @JoinColumn(name="headPolice_id")
    private Police headPolice;
}
